src/script/main.ts

```diff
@@ -1,11 +1,16 @@
+import { stat } from "fs";
+
 export class Main {
     public static viPath;
+    public static viModel;
 
     public static LoadFile(filePath: string) {
-        const currentState = history.state;
+        const currentState = top.history.state;
         const search = "?q=" + filePath;
         if (currentState !== null) {
             const bdDiv: HTMLElement = document.getElementById("BlockDiagram");
+            const stateObj = {hashValue: currentState.filePath.split("#")[1], sourceModel: currentState.parsedContent};
```

Comment:
Minor: Move these new lines of code to either before or after the code that's using `bdDiv`. Let's not add code in the middle of something unrelated to what we're trying to do.

Reply:
Done